### ISSUE TITLE
Segfaults from applications using libsord on nixos-unstable starting at nixos-21.03pre261422.1a57d96edd1

### ISSUE THREAD

**Issue by equill** [NixOS/nixpkgs#109729]
Audio applications depending on `libsord` consistently produce the following error:
```
segfault at 8 ip 00007f9b7d8a0791 sp 00007ffc45471100 error 4 in libsord-0.so.0.16.4[7f9b7d89e000+5000]
```
The hex strings vary, but the rest is consistent. It's possible that other types of app are involved, but these are the only ones I identified, after trying to diagnose the problem with Ardour.

**To Reproduce**
1. Rebuild Nixos from the first-failing release: `sudo nixos-rebuild -I nixpkgs=https://releases.nixos.org/nixos/unstable/nixos-21.03pre261422.1a57d96edd1/nixexprs.tar.xz switch`
2. Attempt to run any of the applications that depend on `libsord`: musE, Guitarix, Qtractor or Ardour. Note that Ardour doesn't crash immediately, but only after configuring its audio options, because that's when it gets to the point of using the library.
3. When it crashes, check the output of `dmesg`.

**Expected behavior**
These applications should run normally, as they do up until nixos-21.03pre260787.56bb1b0f7a3

**Screenshots**
In lieu of screenshots, this is what I get in the output of `dmesg` for each of these applications:
```
[Mon Jan 18 06:43:00 2021] ArdourGUI[1685]: segfault at 8 ip 00007fbb2ac9c791 sp 00007fffc0186db0 error 4 in libsord-0.so.0.16.4[7fbb2ac9a000+5000]
[Mon Jan 18 06:59:59 2021] .muse3-wrapped[2417]: segfault at 8 ip 00007f9b7d8a0791 sp 00007ffc45471100 error 4 in libsord-0.so.0.16.4[7f9b7d89e000+5000]
[Mon Jan 18 07:06:00 2021] .qtractor-wrapp[5655]: segfault at 8 ip 00007f4472118791 sp 00007ffc9bc8f640 error 4 in libsord-0.so.0.16.4[7f4472116000+5000]
[Mon Jan 18 07:08:27 2021] .guitarix-wrapp[5735]: segfault at 8 ip 00007fb68d7de791 sp 00007fffbeb5f3e0 error 4 in libsord-0.so.0.16.4[7fb68d7dc000+5000]
```

**Additional context**
It's possible that `musnix` is a factor here, however, I can reliably produce this failure with nixos-21.03pre261422.1a57d96edd1 or later, and reliably _not_ get it from nixos-21.03pre260787.56bb1b0f7a3 or earlier, with no change to my configs.

`git log 56bb1b0f7a3..1a57d96edd1` combined with `grep` doesn't show anything mentioning `libsord` or any of these applications, so it's most likely a side-effect of a commit not directly related to these things.

I've confirmed that `nixos-20.09` is _not_ affected, and that it's both persistent and consistent across reboots.

**Notify maintainers**

Will come back to this once I've finished trying to bisect it down to the exact commit.

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.9.16, NixOS, 21.03pre260787.56bb1b0f7a3 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixos-21.03pre263901.68398d2dd50"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```
# a list of nixpkgs attributes affected by the problem
attribute:
- ardour
- guitarix
- muse
- qtractor

# a list of nixos modules affected by the problem
module:
```

**Comment by kanashimia** [comment]
@equill it was fixed in #110612, please close this.

**Comment by equill** [comment]
Thanks @kanashimia - much appreciated.
Closing.
